apps/app/src/node/electron-main.ts

```diff
@@ -411,75 +391,91 @@ fontHelper.registerEvents();
 
 let editingStandardInput = false;
 
-ipcMain.on(events.SET_EDITING_STANDARD_INPUT, (event, arg) => {
+ipcMain.on(events.SET_EDITING_STANDARD_INPUT, (_event, arg) => {
   editingStandardInput = arg;
   console.log('Set SET_EDITING_STANDARD_INPUT', arg);
 });
 
+ipcMain.on('FRONTEND_READY', (event) => {
+  const webContents = event.sender;
+  const webContentsId = webContents.id;
+
+  // Check if we have a pending file for the window that just became ready
+  if (pendingFileMap.has(webContentsId)) {
+    const filePath = pendingFileMap.get(webContentsId);
+
+    console.log(`Renderer ${webContentsId} is ready. Sending file: ${filePath}`);
+
+    // Send the file path
+    webContents.send('open-file', filePath);
+
+    // Remove the file from the map so we don't send it again
+    pendingFileMap.delete(webContentsId);
+  }
```

Comment:
The `pendingFileMap` now is only used for the `fileToOpenOnLaunch` for the `welcomeTab` and it is a little bit 雞肋 now.
I think this could be simplified to
```
// Check if we have a pending file for the window that just became ready
  if (webContentsId === tabManager?.welcomeTabId) {
    console.log(`WelcomeTab Id: ${webContentsId} is ready. Sending file: ${fileToOpenOnLaunch}`);

    // Send the file path
    webContents.send('open-file', fileToOpenOnLaunch);

    // Clear the global variable
    fileToOpenOnLaunch = null;
  }
```
and remove `openInitialFileInTab` and `pendingFileMap`